### PR TITLE
Update data loading documentation

### LINKED ISSUE #365: Add documentation of the data loading tutorial

### PR DESCRIPTION
**Pull request by janmiderback**
Updates to the _Data Loading_ tutorials, now also covering how to use load scripts directly instead of using halyard.js. The _Simple Files_ section covers some basic aspects of such load scripts.

**Status**: Dev has reviewed. Documentation need to review.

- [X] **Local files** section
- [X] **Remote files** section
- [X] **Databases** section

Closes #365

### CODE REVIEW

File: docs/tutorials/data-loading/simple-files.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# Simple files
+
+This tutorial gives examples of how to write scripts to load data from simple files such as CSV and Excel files into
+Qlik Associative Engine. It also shows some basic capabilities of load scripts.
+
+!!! Note
+    [halyard.js](https://github.com/qlik-oss/halyard.js) provides functionality to load different kinds of data
+    into the engine without having to write load scripts. The tutorial however, focuses on direct use+ of load scripts
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/simple-files.md]: engine => Qlik Associative Engine

Comment (StefanEnberg) [inline docs/tutorials/data-loading/simple-files.md]: typo use+

File: docs/tutorials/data-loading/simple-files.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+[airports.csv](https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.csv) file into Qlik Associative
+Engine.
+
+Script characteristics:
+
+* Names the resulting table `Airports`
+* Loads data from all headers (`*` notation)
+* Specifies that the file is in text format and `utf8` encoded
+* Sets the value delimiter to the `,` character
+* Uses embedded lables from the CSV file itself as field names
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/simple-files.md]: Maybe add that it by defaults reads them from the first line

File: docs/tutorials/data-loading/databases.md [hunk 3]
```diff
@@ -1,26 +1,33 @@
-# Data Loading
+# Databases
```
Comment (janmiderback) [inline docs/tutorials/data-loading/databases.md]: Don't review this yet.

File: docs/tutorials/data-loading/local-files.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+# Local files
```
Comment (janmiderback) [inline docs/tutorials/data-loading/local-files.md]: @StefanEnberg PTAL at these updates

File: docs/tutorials/data-loading/local-files.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+[load-xlsx-file](https://github.com/qlik-oss/core-data-loading/blob/master/scripts/load-xlsx-file) load script to load
+the same airport data from the
+[airports.xlsx]((https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.xlsx)) Excel file instead.
+
+Script characteristics:
+
+* Names the resulting table `Airports`
+* Loads data from all headers (`*` notation)
+* Specifies that the file is in in Excel format (`ooxml`)
+* Uses embedded lables from the CSV file itself as field names
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/local-files.md]: CSV => xlsx

File: docs/tutorials/data-loading/local-files.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+The expected output is the list of all airports just as in the previous example.
+
+### Subset of fields
+
+This example shows how the
+[load-subset-of-fields](https://github.com/qlik-oss/core-data-loading/blob/master/scripts/load-subset-of-fields) script
+only loads a subset of the fields from the
+[airports.csv](https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.csv) file.
+
+Using the `*` notation in the `LOAD` statement is convenient when loading all data but sometimes you may want to only
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/local-files.md]: Think we should use fields instead of data (might be confusing with the amount of rows

File: docs/tutorials/data-loading/local-files.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+The script achieves this buy _not_ using the `*` notation and instead explicitly lists the headers from the CSV file to
+load into the engine as fields of the same name. In this example, the fields loaded are `Airport`, `City`, `Country`,
+and `TimeZone`.
+
+```sh
+npm start load-subset-of-fields
+```
+
+The expected output is the list of all airports but only the specified fields available, the other fields have not been
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/local-files.md]: but => with

File: docs/tutorials/data-loading/local-files.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+loaded into engine memory.
+
+### Renaming fields
+
+This example shows how the
+[load-renamed-fields](https://github.com/qlik-oss/core-data-loading/blob/master/scripts/load-renamed-fields) renames
+some fields and does not use the header names in the data file.
```
Comment (StefanEnberg) [inline docs/tutorials/data-loading/local-files.md]: Might be a bit confusing. We are still using the header names to define what the source field name will be but then we are aliasing it to another name. If we didn't use embedded labels then each field would be named @1, @2 and so on

File: docs/tutorials/data-loading/remote-files.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+# Remote files
```
Comment (janmiderback) [inline docs/tutorials/data-loading/remote-files.md]: @StefanEnberg Please, also review updates in this file

Comment (StefanEnberg) [reply docs/tutorials/data-loading/remote-files.md]: It looked good 👍

File: docs/tutorials/data-loading/local-files.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+[load-xlsx-file](https://github.com/qlik-oss/core-data-loading/blob/master/scripts/load-xlsx-file) load script to load
+the same airport data from the
+[airports.xlsx]((https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.xlsx)) Excel file instead.
+
+Script characteristics:
+
+* Names the resulting table `Airports`
+* Loads data from all headers (`*` notation)
+* Specifies that the file is in in Excel format (`ooxml`)
+* Uses embedded lables from the Excel file itself as field names
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: labels

File: docs/tutorials/data-loading/local-files.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+[airports.csv](https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.csv) file into Qlik Associative
+Engine.
+
+Script characteristics:
+
+* Names the resulting table `Airports`
+* Loads data from all headers (`*` notation)
+* Specifies that the file is in text format and `utf8` encoded
+* Sets the value delimiter to the `,` character
+* Uses embedded lables from the CSV file itself as field names, reading labels from the first line by default
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: labels

File: docs/tutorials/data-loading/local-files.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+
+```sh
+npm start load-csv-file
+```
+
+The exepected output is a list of all airports loaded with field names as given by the headers in the CSV file. Also
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: expected

File: docs/tutorials/data-loading/local-files.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+The expected output is the list of all airports with only the specified fields available, the other fields have not been
+loaded into engine memory.
+
+### Renaming fields
+
+This example shows how the
+[load-renamed-fields](https://github.com/qlik-oss/core-data-loading/blob/master/scripts/load-renamed-fields) renames
+some fields by providing aliases to the header names.
+
+The script achieves this by using using the `AS` keyword in the `LOAD` statement. In this example, the `rowID` header is
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: remove one using

File: docs/tutorials/data-loading/local-files.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+
+Now install dependencies:
+
+```sh
+npm install
+```
+
+## Sample data files
+
+The examples use these files to load data from:
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: I would rephrase that sentence. Pehaps something like: 
"Following files with data is used in the examples"

File: docs/tutorials/data-loading/local-files.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+
+* [airports.csv](https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.csv) -
+  Headers on first row, comma-separated data
+* [airports.xlsx](https://github.com/qlik-oss/core-data-loading/blob/master/data/airports.xlsx) -
+  Headers on first row, MS Excel Open XML
```
Comment (carlioth) [inline docs/tutorials/data-loading/local-files.md]: Would it be interesting to show how you would do if the header is on the second row, or something diverging from the first example file?

Comment (janmiderback) [reply docs/tutorials/data-loading/local-files.md]: @StefanEnberg and I discussed this and we concluded that we should add such more special cases later if we get feedback on it, and keep it simple for now. You think it is required?

File: docs/tutorials/data-loading/local-files.md (hunk 11 above)
Comment (janmiderback) [reply docs/tutorials/data-loading/local-files.md]: Not sure I undersrand the comment. Header labels result in QAE field names.

File: docs/tutorials/data-loading/local-files.md (hunk 10 above)
Comment (janmiderback) [reply docs/tutorials/data-loading/local-files.md]: Same here. In QAE it results in field names.

**Comment by janmiderback** [comment]
@StefanEnberg @carlioth PTAL at the updated Database section.

File: docs/tutorials/data-loading/local-files.md (hunk 10 above)
Comment (carlioth) [reply docs/tutorials/data-loading/local-files.md]: Sorry, it says 'lables' and should be labels

File: docs/tutorials/data-loading/local-files.md (hunk 11 above)
Comment (carlioth) [reply docs/tutorials/data-loading/local-files.md]: Should be more clear, says 'lables' should be labels

**Comment by qlikossbuild** [comment]
Deploy preview for *qlikcore* ready!

Built with commit 1a2dc3c817b12f99b96ae5b5632a42fc1200da20

https://deploy-preview-366--qlikcore.netlify.com

File: docs/tutorials/data-loading/databases.md [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+const connectionId = await app.createConnection(connectionSettings);
+```
+
+In the `connectionSettings` object:
+
+* `qType` represents what type of connector the connection should use.
+* `qName` is the name of this connection instance.
+* `qConnectionString` is the parameter that is sent to the connector. The parts of the connection string that is
+  specific to the JDBC connector is the driver setting. It will be the JDBC driver that the connector will use to
+  connec to the database with. The provider needs to be the same as the `qType`. Host, port and database are related to
```
Comment (carlioth) [inline docs/tutorials/data-loading/databases.md]: spelling - connect

File: docs/tutorials/data-loading/databases.md [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+npm start
+```
+
+The expected output is that 100 rows of airport data fetched from MySQL and an additional 100 rows of data
+fetched from PostgreSQL are printed to the console.
+
+## What is happening?
+
+Once the containers are running and you trigger the reload, the program creates and opens an app called `reloadapp.qvf`
+on the Qlik Associative Engine. Then it creates a two connections of the type we defined earlier to load data from both
```
Comment (carlioth) [inline docs/tutorials/data-loading/databases.md]: Remove "a" in "a two connections"

File: docs/tutorials/data-loading/databases.md [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+with JDBC connections, in this way working as a data bridge between the engine and the data sources.
+
+The Qlik Associative Engine container must be configured to use the custom JDBC connector by providing some command-line
+options to it. Study the
+[docker-compose.yml](https://github.com/qlik-oss/core-grpc-jdbc-connector/blob/master/example/docker-compose.yml) file
+and notice the following options:
+
+* `-S EnableGrpcCustomConnectors=1` which enables gRPC connectors in the Qlik Associative Engine.
+* `-S GrpcConnectorPlugins="jdbc,jdbc-connector:50051"` which registers a connector of the type `jdbc`, which we tell
+  the Qlik Associative Engine exists on host `jdbc-connector` and listening on port 50051. (`jdbc` is here an arbitrary
```
Comment (carlioth) [inline docs/tutorials/data-loading/databases.md]: I'm not sure if it's the first part of "jdbc,jdbc-connector:50051" or the second that jdbc is arbitrary.

Comment (janmiderback) [reply docs/tutorials/data-loading/databases.md]: It is the first one before the comma. I'll clarify that :+1:
